Tools/CISupport/ews-build/steps.py

```diff
@@ -1463,7 +1463,33 @@ class UnApplyPatchIfRequired(CleanWorkingDirectory):
     descriptionDone = ['Unapplied patch']
 
     def doStepIf(self, step):
-        return self.getProperty('patchFailedToBuild') or self.getProperty('patchFailedTests')
+        return self.getProperty('patch_id') and (self.getProperty('changeFailedToBuild') or self.getProperty('changeFailedTests'))
+
+    def hideStepIf(self, results, step):
+        return not self.doStepIf(step)
+
+
+class RevertPullRequestIfRequired(steps.ShellSequence):
+    name = 'revert-pull-request'
```

Comment:
I wonder if 'revert-pull-request' is appropriate name. Can a pull request really be reverted? We are reverting few commits on that branch.

Reply:
I don't know a better name.

You are basically checking out the commit your pull request originated from. I'm not against a better name, just not sure what it would be.

Reply:
Maybe let's call it revert-pull-request-changes for now.